src/wazuh_modules/vulnerability_scanner/src/vulnerabilityScannerFacade.cpp

```diff
@@ -214,95 +216,138 @@ void VulnerabilityScannerFacade::initWazuhDBEventSubscription()
+        // If the scanner was enabled on the manager, perform a re-scan
+        else if (managerLastState.compare(DISABLED) == 0 && managerState.compare(ENABLED) == 0)
         {
-            logInfo(WM_VULNSCAN_LOGTAG, "Vulnerability scanner in manager activated. Performing scan.");
-
-            // Create and queue the scan event
-            nlohmann::json dataValueJson;
-            dataValueJson["action"] = "scanAgent";
-            dataValueJson["agent_info"]["agent_id"] = "000";
-
-            const auto& dataValue = dataValueJson.dump();
-            const std::vector<char> message(dataValue.begin(), dataValue.end());
-            pushEvent(message, BufferType::BufferType_JSON);
+            logDebug1(WM_VULNSCAN_LOGTAG, "Vulnerability scanner in manager enabled (manager re-scan needed)");
+            m_managerAction = ActionWrapper::Action::SCAN;
         }
     }
-
-    stateDB.put(VD_MANAGER_STATE_KEY, managerScanStatus == ManagerScanStatus::DISABLED ? DISABLED : ENABLED);
+    stateDB.put(VD_MANAGER_STATE_KEY, managerState);
 }
 
 void VulnerabilityScannerFacade::clusterConfigurationChange(Utils::RocksDBWrapper& stateDB) const
 {
-    const auto& clusterName = PolicyManager::instance().getClusterName();
-
+    // Check cluster name changes
+    const auto clusterName = PolicyManager::instance().getClusterName();
     if (std::string clusterLastName; stateDB.get(CLUSTER_NAME_KEY, clusterLastName))
     {
         // If the cluster name changed, perform a scan
         if (clusterLastName.compare(clusterName) != 0)
         {
             logDebug1(WM_VULNSCAN_LOGTAG, "Cluster name changed (re-scan needed)");
-            m_shouldRescan.store(true);
+            m_agentsAction = ActionWrapper::Action::SCAN;
         }
     }
-
     stateDB.put(CLUSTER_NAME_KEY, clusterName);
+
+    // Check cluster state changes
+    const auto clusterState = PolicyManager::instance().getClusterStatus() ? ENABLED : DISABLED;
+    if (std::string clusterLastState; stateDB.get(CLUSTER_STATE_KEY, clusterLastState))
+    {
+        // If the cluster status changed, perform a scan
+        if (clusterLastState.compare(clusterState) != 0)
+        {
+            logDebug1(WM_VULNSCAN_LOGTAG, "Cluster was enabled/disabled (agents re-scan needed)");
+            m_agentsAction = ActionWrapper::Action::SCAN;
+        }
+    }
+    stateDB.put(CLUSTER_STATE_KEY, clusterState);
+
+    // Check cluster node name changes
+    const auto& clusterNodeName = PolicyManager::instance().getClusterNodeName();
+    if (std::string clusterNodeLastName; stateDB.get(CLUSTER_NODE_NAME_KEY, clusterNodeLastName))
+    {
+        // If the cluster node name changed, perform a scan
+        if (clusterNodeLastName.compare(clusterNodeName) != 0)
+        {
+            logDebug1(WM_VULNSCAN_LOGTAG, "Cluster node name changed (manager re-scan needed)");
+            m_managerAction = ActionWrapper::Action::SCAN;
```

Comment:
Do we need to check `managerDisabledScan` here?

Reply:
I've added a new condition